drivers/flash/Kconfig.stm32

```diff
@@ -11,7 +11,6 @@ config SOC_FLASH_STM32
 	depends on DT_HAS_ST_STM32_FLASH_CONTROLLER_ENABLED
 	select FLASH_HAS_DRIVER_ENABLED
 	default y
-	select FLASH_PAGE_LAYOUT
```

Comment:
This is also done in few other files such as: 
./drivers/flash/Kconfig.sam
./drivers/flash/Kconfig.stm32_ospi
./drivers/flash/Kconfig.stm32_qspi

Do you want to clean them as well ?

Reply:
Yes. I have missed them.

Reply:
Fixed.